src/test/java/com/wikia/webdriver/testcases/auth/ForgottenPasswordTests.java

```diff
@@ -0,0 +1,68 @@
+package com.wikia.webdriver.testcases.auth;
+
+import com.wikia.webdriver.common.core.Assertion;
+import com.wikia.webdriver.common.core.MailFunctions;
+import com.wikia.webdriver.common.core.configuration.Configuration;
+import com.wikia.webdriver.common.core.helpers.User;
+import com.wikia.webdriver.common.properties.Credentials;
+import com.wikia.webdriver.common.templates.NewTestTemplate;
+import com.wikia.webdriver.pageobjectsfactory.componentobject.global_navitagtion.NavigationBar;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.auth.ResetPasswordPage;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.auth.signin.AttachedSignInPage;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.WikiBasePageObject;
+
+import com.wikia.webdriver.pageobjectsfactory.pageobject.auth.signin.DetachedSignInPage;
+import org.testng.annotations.Test;
+
+@Test(groups = "auth-forgottenPassword")
+public class ForgottenPasswordTests extends NewTestTemplate {
+
+  Credentials credentials = Configuration.getCredentials();
+
+  @Test(groups = "ForgottenPassword_anonCanRemindPasswordFromAuthModal")
+  public void anonCanRemindPasswordFromAuthModal() {
+        MailFunctions.deleteAllEmails(credentials.email, credentials.emailPassword);
+    WikiBasePageObject base = new WikiBasePageObject();
+    base.openWikiPage(wikiURL);
+    DetachedSignInPage loginModal = new DetachedSignInPage(new NavigationBar(driver).clickOnSignIn());
+    loginModal
+      .clickForgotPasswordLink()
+      .requestLinkForUsername(User.FORGOTTEN_PASSWORD.getUserName());
+
+    String resetLink = base.getPasswordResetLink(credentials.email, credentials.emailPassword);
+    ResetPasswordPage resetPass = new ResetPasswordPage(resetLink);
+    resetPass.setNewPassword(User.FORGOTTEN_PASSWORD.getPassword());
+
+    Assertion.assertTrue(resetPass.newPasswordSetSuccessfully());
+
+  }
+
+  @Test(groups = "ForgottenPassword_anonCanRemindPasswordOnUserLoginSpecialPage")
+  public void anonCanRemindPasswordOnUserLoginSpecialPage() {
+    MailFunctions.deleteAllEmails(credentials.email, credentials.emailPassword);
+    WikiBasePageObject base = new WikiBasePageObject();
+    AttachedSignInPage signIn = new AttachedSignInPage().open();
+    signIn.clickForgotPasswordLink().requestLinkForUsername(User.FORGOTTEN_PASSWORD.getUserName());
+    String resetLink = base.getPasswordResetLink(credentials.email, credentials.emailPassword);
+    ResetPasswordPage resetPass = new ResetPasswordPage(resetLink);
+    resetPass.setNewPassword(User.FORGOTTEN_PASSWORD.getPassword());
+
+    Assertion.assertTrue(resetPass.newPasswordSetSuccessfully());
+
+  }
```

Comment:
Seems to be you're doing three things:
1. setup (deleting old mails - see other comment)
1. going to forgot password page and requesting new password
1. going to the page from e-mail and setting new password
1. verifying the new password has been set up correctly

Could you maybe refactor this tests into separate methods which names indicate those steps? You don't have to have new methods for one liners and where it's clearly indicated what's happening, but in places it makes sense, for example:
```
private void setNewPasswordAndAssertSuccess(String resetLink) {
    ResetPasswordPage resetPass = new ResetPasswordPage(resetLink);
    resetPass.setNewPassword(User.FORGOTTEN_PASSWORD.getPassword());
    Assertion.assertTrue(resetPass.newPasswordSetSuccessfully());
}
```

Also you may want to reorder some of the lines, for example base object is initialised but used only a few lines later.